css.js

```diff
@@ -1,36 +1,92 @@
 var loader = require("@loader");
 
+var isNode = typeof process === "object" && {}.toString.call(process) ===
+	"[object process]";
+var importRegEx = /@import [^uU]['"]?([^'"\)]*)['"]?/g;
+var resourceRegEx =  /url\(['"]?([^'"\)]*)['"]?\)/g;
+	
+var waitSeconds = 100;
+var noop = function () {};
+var isWebkit = (typeof window !== 'undefined') ?
+	window.navigator.userAgent.match(/AppleWebKit\/([^ ;]*)/) : false;
+var webkitLoadCheck = function (link, callback) {
+		setTimeout(function () {
+			for (var i = 0; i < document.styleSheets.length; i++) {
+				var sheet = document.styleSheets[i];
+				if (sheet.href == link.href)
+					return callback();
+			}
+			webkitLoadCheck(link, callback);
+		}, 10);
+	};
+
 if(isProduction()) {
 	exports.fetch = function(load) {
-		// return a thenable for fetching (as per specification)
-		// alternatively return new Promise(function(resolve, reject) { ... })
-		var cssFile = load.address;
+		// stolen from https://github.com/systemjs/plugin-css/edit/master/css.js
+		// thanks!
+		// wait until the css file is loaded
+		return new Promise(function(resolve, reject) {
+			var timeout = setTimeout(function() {
+				reject('Unable to load CSS');
+			}, waitSeconds * 1000);
 
-		var link = document.createElement('link');
-		link.rel = 'stylesheet';
-		link.href = cssFile;
+			var _callback = function(error) {
+				clearTimeout(timeout);
+				link.onload = link.onerror = noop;
+				setTimeout(function() {
```

Comment:
Why is this setTimeout needed? this callback is called after the link is loaded, can't we just go ahead and resolve/reject() immediately?